scilpy/tractanalysis/scoring.py

```diff
@@ -52,38 +60,66 @@ def compute_dice_overlap_overreach(current_vb_voxels, gt_mask, dimensions):
         The voxels touched by at least one streamlines for a given bundle.
     gt_mask: 3D array
         The ground truth mask.
-    dimensions: array
+    dimensions: np.array
         The nibabel dimensions of the data (3D).
 
     Returns
     -------
     dice: float
         The dice score
-    overlap: int
-        The overlap (in number of voxels, not as percentages).
-    overreach: int
-        The overreach (in number of voxels).
-    lacking: int
-        The number of voxels from the gt_mask that have not been recovered.
+    tp_nb_voxels: int
+        The TP (true positive) count in number of voxels.
+    fp_nb_voxels: int
+        The FP (false positive) count in number of voxels.
+        Hint: Divide it by the ground truth count to get the overreach, or
+        by the recovered bundle count to get the ORn (scores used in the
+        ismrm2015 tractography challenge).
+    fn_nb_voxels: int
+        The number of voxels from the gt_mask that have not been recovered;
+        corresponds to the FN count (false negative).
+    overlap: float
+        TP divided by the ground truth count (i.e. TP + FN), in percentage.
+    overreach_pct_total: float
+        The overreach, normalized by the recovered bundle's area.
+    overreach_pct_gt: float
+        The overreach, normalized by the ground truth area.
     """
-    # Dice
-    dice = compute_dice_voxel(gt_mask, current_vb_voxels)[0]
+    # True positive = |B inter A|
+    tp_mask = gt_mask * current_vb_voxels
+    tp_nb_voxels = np.count_nonzero(tp_mask)
 
-    # Overlap and overreach
-    overlap_mask = gt_mask * current_vb_voxels
-    overreach_mask = np.zeros(dimensions)
-    overreach_mask[np.where(
+    # False positive = |B except A|
+    fp_mask = np.zeros(dimensions)
+    fp_mask[np.where(
         (gt_mask == 0) & (current_vb_voxels >= 1))] = 1
+    fp_nb_voxels = np.count_nonzero(fp_mask)
 
-    bundle_lacking = np.zeros(dimensions)
-    bundle_lacking[np.where(
+    # False negative = |A except B|
+    fn_mask = np.zeros(dimensions)
+    fn_mask[np.where(
         (gt_mask == 1) & (current_vb_voxels == 0))] = 1
+    fn_nb_voxels = np.count_nonzero(fn_mask)
+
+    gt_total_nb_voxels = tp_nb_voxels + fn_nb_voxels
+    # Same as np.count_nonzero(gt_mask)
+
+    nb_voxels_total = tp_nb_voxels + fp_nb_voxels
+    # Same as np.count_nonzero(current_vb_voxels)
 
-    overlap = np.count_nonzero(overlap_mask)
-    overreach = np.count_nonzero(overreach_mask)
-    lacking = np.count_nonzero(bundle_lacking)
+    # Overlap = |B inter A| / |A|
+    overlap = tp_nb_voxels / gt_total_nb_voxels
+
+    # Overreach: two versions are sometimes used.
+    # |B except A| / |A| or |B except A| / |B|
+    overreach_pct_total = fp_nb_voxels / nb_voxels_total
+    overreach_pct_gt = fp_nb_voxels / gt_total_nb_voxels
+
+    # Dice and f1. Should be the same.
+    dice = compute_dice_voxel(gt_mask, current_vb_voxels)[0]
+    f1 = compute_f1_score(overlap, overreach_pct_total)
```

Comment:
Why include both if they're the same ?

Reply:
I have no idea. It was like that when I first started working on the code. I didn't want to remove it in case I was wrong. Do you think I should remove it?

Reply:
Yes, I don't see the need for both. Thanks !